internal/baseservice/base_service.go

```diff
@@ -143,24 +143,27 @@ func (s *BaseService) CancellableSleepRandomBetweenC(ctx context.Context, sleepD
 // another value instead, but shouldn't unless there's a good reason to do so.
 const MaxAttemptsBeforeResetDefault = 10
 
-// CancellableSleepExponentialBackoff sleeps for a reasonable exponential
-// backoff interval for a service based on the given attempt number. Uses a 2**N
-// second algorithm, +/- 10% random jitter. Sleep is cancelled if the given
-// context is cancelled.
-func (s *BaseService) CancellableSleepExponentialBackoff(ctx context.Context, attempt, maxAttemptsBeforeReset int) {
-	s.CancellableSleep(ctx, timeutil.SecondsAsDuration(s.exponentialBackoffSeconds(attempt, maxAttemptsBeforeReset)))
-}
-
-func (s *BaseService) exponentialBackoffSeconds(attempt, maxAttemptsBeforeReset int) float64 {
+// ExponentialBackoff returns a duration for a reasonable exponential backoff
+// interval for a service based on the given attempt number, which can then be
+// fed into CancellableSleep to perform the sleep. Uses a 2**N second algorithm,
+// +/- 10% random jitter. Sleep is cancelled if the given context is cancelled.
+//
+// Attempt should start at one for the first backoff/failure.
+func (s *BaseService) ExponentialBackoff(attempt, maxAttemptsBeforeReset int) time.Duration {
```

Comment:
I retooled this a bit so it returns a duration instead of doing the sleep because it makes it easier to include the sleep duration in log statements which is pretty nice.